### DISCUSSION TITLE
Changing theme with realtime changes

### DISCUSSION THREAD

**Discussion by ghost** [TomSchimansky/CustomTkinter#1667] [Q&A]
Hi!

I'm using CustomTkinter 5.1.3 and I was wondering if there was any way to change the color theme and have all the widgets update in realtime. For example, when you change the appearance mode with `.set_appearance_mode()`, you see the whole application and every single widget update. I want to do the same, but with color themes. I'm currently using `.set_default_color_theme()` but that only works if I redraw/grid/create  the widgets again after calling the method.

Any ideas?

**Comment by avalon60** [comment]
Perhaps the only way you could do this, is by keeping a track of created widgets in a list, and using the list to tear them down, and then you would need to rebuild them. The problem is that you might lose session state, depending on your app design.

**Reply by ghost** [reply]
So there's no way to change theme in realtime _without_ destroying the whole application...

Hm, okay, thank you.

**Comment by ghost** [comment]
I selected my own comment as the answer because it was the only one stating that there is no way to achieve what I want.

**Reply by avalon60** [reply]
Yes @dishb, but you answer wasn't strictly correct. It depends on the app design / functional requirements ("you might"). 😏

**Reply by ghost** [reply]
oh no - unmarked answer 😅
